bundles/org.openhab.binding.mqtt.generic/src/main/resources/OH-INF/config/switch-channel-config.xml

```diff
@@ -79,15 +79,25 @@
 			<advanced>true</advanced>
 		</parameter>
 
-		<parameter name="on" type="text">
-			<label>Custom On/Open Value</label>
-			<description>A number (like 1, 10) or a string (like "enabled") that is additionally recognised as on/open state. You
-				can use this parameter for a second keyword, next to ON (OPEN respectively on a Contact).</description>
+		<parameter name="onState" type="text" multiple="true">
+			<label>Custom On/Open Values</label>
+			<description>A list of numbers (like 1, 10) or strings (like "enabled") which are additionally recognised as on/open
+				state. You
+				can use this parameter for additional keywords, next to ON (OPEN respectively on a Contact).</description>
 		</parameter>
-		<parameter name="off" type="text">
-			<label>Custom Off/Closed Value</label>
-			<description>A number (like 0, -10) or a string (like "disabled") that is additionally recognised as off/closed
-				state. You can use this parameter for a second keyword, next to OFF (CLOSED respectively on a Contact).</description>
+		<parameter name="offState" type="text" multiple="true">
+			<label>Custom Off/Closed Values</label>
+			<description>A list of numbers (like 0, -10) or strings (like "disabled") which are additionally recognised as
+				off/closed
+				state. You can use this parameter for additional keywords, next to OFF (CLOSED respectively on a Contact).</description>
+		</parameter>
+		<parameter name="on" type="text" advanced="true">
```

Comment:
why rename the parameter? I know it makes it somewhat clearer, but it also means people will need to update their configuration at some point. my understanding from @jimtng's investigations are that you can add `multiple="true"` to an existing parameter, and then the core framework will automatically handle always presenting it to the binding as a list, regardless of if it's actually a list in the persisted configuration (either json or .things file)

Reply:
The problem is, that I would need to change the type of the `on` and `off` params in the `ChannelConfig` class from `String` to `List<String>`, which would require changes to several types, whilst changes to the `onState` and `offState` param require less changes.

Reply:
This makes it really difficult IMO to implement the changes I want to do in a "clean" way - FYI I have already changed my approach and actually don't need the changes from this PR at all. Just use a SCRIPT transformation on the Item channel link and you can do whatever you want ...

Reply:
> FYI I have already changed my approach and actually don't need the changes from this PR at all.

What's your plan for this PR? continue or drop ?

Reply:
I would tend to drop, as it is just way easier to evaluate the string with a script transformation ...
I will close this PR then.